### ISSUE TITLE
Got invalid conversion from 'void*' to 'uint8_t* 

### ISSUE THREAD

**Issue by openmtl** [nategri/nematoduino#2]
Using Arduino IDE 1.6.7 on Ubuntu 14.04. I did a git clone and then used Arduino IDE verify to build and got error messages about invalid conversion from 'void*' to 'uint8_t* {aka unsigned char*}' [-fpermissive] for the 3 malloc  for CurrMuscleState, NextMuscleState and IdleCycles so I cast the void to the pointer type it wanted. It compiles.

My Arduino has been collecting dust but want to make it do something and this idea of the neural net is interesting.  The diff is as follows. 

```
diff --git a/nemotoduino.ino b/nemotoduino.ino
index 631b66f..8cd3968 100644
--- a/nemotoduino.ino
+++ b/nemotoduino.ino
@@ -39,11 +39,11 @@ int8_t CurrConnectedState[N_NTOTAL];
 int8_t NextConnectedState[N_NTOTAL];
 
 // Another set for muscles that aren't connected to other cells
-int16_t* CurrMuscleState = malloc((N_NTOTAL-N_MAX)*sizeof(int16_t));
-int16_t* NextMuscleState = malloc((N_NTOTAL-N_MAX)*sizeof(int16_t));
+int16_t* CurrMuscleState = (int16_t*)malloc((N_NTOTAL-N_MAX)*sizeof(int16_t));
+int16_t* NextMuscleState = (int16_t*)malloc((N_NTOTAL-N_MAX)*sizeof(int16_t));
 
 // Final set to track how many cycles a neuron has been idle
-uint8_t* IdleCycles = malloc(N_MAX*sizeof(uint8_t));
+uint8_t* IdleCycles = (uint8_t*)malloc(N_MAX*sizeof(uint8_t));
 
 //
 // Functions for getting and setting these states
```

**Comment by EdSchenk** [comment]
I changed the code to:

int16_t* CurrMuscleState = (int16_t*)malloc((N_NTOTAL - N_MAX) * sizeof(int16_t));
int16_t* NextMuscleState = (int16_t*)malloc((N_NTOTAL - N_MAX) * sizeof(int16_t));
uint8_t* IdleCycles = (uint8_t*)malloc(N_MAX*sizeof(uint8_t));

**Comment by openmtl** [comment]
Yup, that's the same as my diff. I've never really worked with git that much. Do you want to do a pull request or me ?  In reading about dynamic memory allocation on embedded chips many seem to shun this and do things like create a buffer but I haven't even got a motor shield yet to test this all out physically so I am overreaching right now.
